src/main/java/nextstep/subway/path/dto/PathFinderResult.java

```diff
@@ -0,0 +1,66 @@
+package nextstep.subway.path.dto;
+
+import nextstep.subway.line.dto.LineResponse;
+import nextstep.subway.line.dto.LineStationResponse;
+import nextstep.subway.station.dto.StationResponse;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class PathFinderResult {
+
+    private final List<Long> stationIds;
+    private final double weight;
+
+    public PathFinderResult(List<Long> stationIds, double weight) {
+        this.stationIds = stationIds;
+        this.weight = weight;
+    }
+
+    public List<Long> getStationIds() {
+        return stationIds;
+    }
+
+    public double getWeight() {
+        return weight;
+    }
+
+    public PathResponse toPathResponse(List<LineResponse> lines) {
+        final Map<Long, StationResponse> stationResponseMap = extractStationResponseMap(lines);
+        final List<StationResponse> list = stationIds.stream().map(stationResponseMap::get).collect(Collectors.toList());
+        final int distance = getDistance(lines, list);
+        final int duration = getDuration(lines, list);
+        return new PathResponse(list, distance, duration);
+    }
+
+    private int getDuration(List<LineResponse> lines, List<StationResponse> list) {
+        return getLineStationResponseStream(lines, list).mapToInt(LineStationResponse::getDuration).sum();
+    }
+
+    private int getDistance(List<LineResponse> lines, List<StationResponse> list) {
+        return getLineStationResponseStream(lines, list).mapToInt(LineStationResponse::getDistance).sum();
+    }
+
+    private Stream<LineStationResponse> getLineStationResponseStream(List<LineResponse> lines, List<StationResponse> list) {
+        return lines.stream()
+                .flatMap(lineResponse -> lineResponse.getStations().stream())
+                .filter(lineStationResponse -> lineStationResponse.getStation().getId() != null)
+                .filter(lineStationResponse -> list.contains(lineStationResponse.getStation()));
+    }
+
+    private Map<Long, StationResponse> extractStationResponseMap(List<LineResponse> lines) {
+        final Map<Long, StationResponse> map = new HashMap<>();
+        final List<StationResponse> list = lines.stream()
+                .flatMap(lineResponse -> lineResponse.getStations().stream())
+                .map(LineStationResponse::getStation)
+                .distinct()
+                .collect(Collectors.toList());
+        for (StationResponse stationResponse : list) {
+            map.put(stationResponse.getId(), stationResponse);
+        }
+        return map;
```

Comment:
stream이 항상 옳은것은 아니지만 아래와 같이 수정해 볼 수 있을 것 같아요 :)
```suggestion
        return lines.stream()
                .flatMap(lineResponse -> lineResponse.getStations().stream())
                .map(LineStationResponse::getStation)
                .distinct()
                .collect(Collectors.toMap(StationResponse::getId, Function.identity()));
```